docarray/array/storage/weaviate/backend.py

```diff
@@ -115,6 +115,30 @@ def _init_storage(
             if isinstance(_docs, Document):
                 self.append(_docs)
 
+    def _init_subindices(self, *args, **kwargs):
+        from docarray import DocumentArray
+
+        self._subindices = {}
+        subindex_configs = kwargs.get('subindex_configs', None)
```

Comment:
why do not u pass it as an argument? so that u can pass proper `docstrings` and all of that?

Reply:
It is because when such a da gets created through the syntax sugar `DocumentArray(storage='weaviate', ...)`, all the *args and **kwargs get passed to the specialized backend class. This is the code snippet:
```python
class BaseDocumentArray(MutableSequence[Document]):
    def __init__(self, *args, storage: str = 'memory', **kwargs):
        super().__init__()
        self._init_storage(*args, **kwargs)
        self._init_subindices(*args, **kwargs)
```
So when `_init_subindices` takes `subindex_configs` as a normal argument, it will mistake the first argument that the user passes as `subindex_configs`. Usually that is the list of Documents that make up the DocumentArray, and `_init_subindices` will think that this list of Documents is `subindex_configs`.

What I am doing here is enforcing `subindex_configs` to be passes with the keyword, not as a positional argument, so that no confusion can happen between args.